.github/workflows/check-state-compatibility.yaml

```diff
@@ -49,39 +52,62 @@ jobs:
 
   check_state_compatibility:
     # DO NOT CHANGE THIS: please read the note above
-    if: ${{ github.event_name == 'push' || github.event.pull_request.head.repo.full_name == 'osmosis-labs/osmosis' }} 
+    if: ${{ github.event_name == 'push' || github.event_name == 'schedule' || github.event.pull_request.head.repo.full_name == 'osmosis-labs/osmosis' }}
     runs-on: osmo-runner
     steps:
       - 
-        name: Checkout repository
+        name: Checkout commit
+        if: ${{ github.event_name != 'schedule' }}
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - 
-        name: Build the osmosisd binary with osmobuilder
-        run: |
-          make -f contrib/images/osmobuilder/Makefile get-binary-amd64
-      - 
-        name: Copy osmosisd binary to PATH and check version
+        # If workflow was triggered by a schedule, 
+        # explicitly checkout the correct branch
+        name: Checkout v10.x branch
+        if: ${{ github.event_name == 'schedule' }}
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: v10.x
+      -
+        name: 🔨 Build the osmosisd binary with osmobuilder
         run: |
           VERSION=$(echo $(git describe --tags) | sed 's/^v//')
-          sudo cp osmosisd-$VERSION-linux-amd64 /usr/local/bin/osmosisd
+          make -f contrib/images/osmobuilder/Makefile get-binary-amd64
+
+          sudo cp release/osmosisd-$VERSION-linux-amd64 /usr/local/bin/osmosisd
           sudo chmod +x /usr/local/bin/osmosisd
           osmosisd version
-        working-directory: release
-      - 
-        name: Download pre-epoch snapshot
+      -
+        name: 🧪 Initialize Osmosis Node
         run:  |
-          rm -rf $HOME/.osmosisd/data
-          SNAPSHOT_URL=$(curl -s ${{ env.SNAPSHOT_URL }}  | dasel --plain -r json  '(file=osmosis-1-pre-epoch).url')
-          wget -q -O - $SNAPSHOT_URL | lz4 -d | tar -C $HOME/.osmosisd/ -xvf - 
-      - 
+          rm -rf $HOME/.osmosisd/ || true
+          osmosisd init osmo-runner -o
+          wget -O $HOME/.osmosisd/config/genesis.json ${{ env.GENESIS_URL }}
+      -
+        name: ⏬ Download pre-epoch snapshot
+        run:  |
+          SNAPSHOT_INFO=$(curl -s --retry 5 --retry-delay 5 --connect-timeout 30 -H "Accept: application/json" ${{ env.SNAPSHOT_URL }})
+          SNAPSHOT_URL=$(echo $SNAPSHOT_INFO | dasel --plain -r json  '(file=osmosis-1-pre-epoch).url')
+          SNAPSHOT=$(echo $SNAPSHOT_INFO | dasel --plain -r json  '(file=osmosis-1-pre-epoch).filename' | cut -f 1 -d '.')
+
+          mkdir -p $HOME/snapshots/
+          if [ ! -d "$HOME/snapshots/$SNAPSHOT" ]; then
```

Comment:
I'm assuming that this `mkdir` and checking if the directory exists logic is due to caching snapshots. Could you please add a comment that we do this because of caching?

Reply:
Added